docs/docs/protocol-specs/gas-and-fees/kernel-tracking.md

```diff
@@ -295,7 +296,7 @@ class PublicKernelCircuitPublicInputs {
   +PublicAccumulatedData end
   +CombinedConstantData constants
   +AztecAddress fee_payer
-  +CallRequest public_teardown_call_request
+  +CallRequest[MAX_PUBLIC_CALL_STACK_LENGTH_PER_TX] public_teardown_call_stack
```

Comment:
We should create an issue to roll this back to be a single item when the AVM ships, since it'll process all call requests in a single run. In other words, that if the teardown function calls nested functions, it'll still result in a single AVM run with a single public kernel iteration wrapping it.

Reply:
I created https://github.com/AztecProtocol/aztec-packages/issues/6305